Index: cli.py
===================================================================
--- cli.py
+++ cli.py	2009-02-18 15:31:28.000000000 +0100
@@ -491,7 +491,7 @@
                             needrepos.append(po.repoid)
 
                     self.log(2, 'Importing Changelog Metadata')
-                    self.repos.populateSack(with='otherdata', which=needrepos)
+                    self.repos.populateSack(include='otherdata', which=needrepos)
                     self.log(2, 'Generating RSS File for %s' % pkgtype)
                         
                     self.listPkgs(this_pkg_list, titles[pkgtype], outputType='rss')
@@ -602,9 +602,9 @@
             self.log(3, '%s' % self.pickleRecipe())
             try:
                 self.doRepoSetup(nosack=1)
-                self.repos.populateSack(with='metadata', pickleonly=1)
-                self.repos.populateSack(with='filelists', pickleonly=1)
-                self.repos.populateSack(with='otherdata', pickleonly=1)
+                self.repos.populateSack(include='metadata', pickleonly=1)
+                self.repos.populateSack(include='filelists', pickleonly=1)
+                self.repos.populateSack(include='otherdata', pickleonly=1)
                 
             except yum.Errors.YumBaseError, e:
                 return 1, [str(e)]
Index: yum/__init__.py
===================================================================
--- yum/__init__.py
+++ yum/__init__.py	2009-02-18 15:32:05.000000000 +0100
@@ -904,7 +904,7 @@
                     matched = 1
             if not matched:
                 self.log(2, 'Importing Additional filelist information for packages')
-                self.repos.populateSack(with='filelists')
+                self.repos.populateSack(include='filelists')
 
         for arg in args:
             restring = self._refineSearchPattern(arg)
Index: yum/depsolve.py
===================================================================
--- yum/depsolve.py
+++ yum/depsolve.py	2009-02-18 15:32:47.000000000 +0100
@@ -67,7 +67,7 @@
                     matched = 1
             if not matched:
                 self.log(2, 'Importing Additional filelist information for dependency resolution')
-                self.repos.populateSack(with='filelists')
+                self.repos.populateSack(include='filelists')
                 
         pkgs = self.pkgSack.searchProvides(name)
         if flags == 0:
Index: yum/repos.py
===================================================================
--- yum/repos.py
+++ yum/repos.py	2009-02-18 15:30:36.000000000 +0100
@@ -202,10 +202,10 @@
             repo.setupGrab()
             
                 
-    def populateSack(self, which='enabled', with='metadata', callback=None, pickleonly=0):
+    def populateSack(self, which='enabled', include='metadata', callback=None, pickleonly=0):
         """This populates the package sack from the repositories, two optional 
            arguments: which='repoid, enabled, all'
-                      with='metadata, filelists, otherdata, all'"""
+                      include='metadata, filelists, otherdata, all'"""
 
         if not callback:
             callback = self.callback
@@ -223,10 +223,10 @@
                 repobj = self.getRepo(which)
                 myrepos.append(repobj)
 
-        if with == 'all':
+        if include == 'all':
             data = ['metadata', 'filelists', 'otherdata']
         else:
-            data = [ with ]
+            data = [ include ]
         
         for repo in myrepos:
             if not hasattr(repo, 'cacheHandler'):
